### DISCUSSION TITLE
How to run a complex/multi shell command task properly?

### DISCUSSION THREAD

**Discussion by rlhk** [babashka/babashka#1443] [Q&A] [answered]
I have a command in package.json script like:
`"rm -fr ./dist ./out/* && yarn && mkdir ./dist && cp resources/* ./dist/ && npx shadow-cljs watch :plugin :test"`

Trying to move to bb :task like `(shell "rm -fr ./dist ./out/* && yarn && mkdir ./dist && cp resources/* ./dist/ && npx shadow-cljs watch :plugin :test")` obvious doesn't work

The guide has no hint. How should it be achieved / organised in bb ? Wrapping it in another shell script seems to defeat bb task's purpose. Multiple `shell` in `do` block?

**Comment by borkdude** [comment, marked as answer]
@rlhk Yes, multiple `shell` in a `do` block.

Something like:

``` clojure
(fs/delete-tree "dist")
(fs/delete-tree "out")
(shell "yarn")
(fs/create-dir "dist")
(fs/copy-tree "resources" "dist")
(shell "npx shadow-cljs watch :plugin :test")
```

There may be a bug in there, but that's the gist of it.

**Reply by rlhk** [reply]
More consistent with shell cmds this way:
```
(doseq [cmd ["rm -fr ./dist ./out/*"
             "yarn", "mkdir ./dist"
             "cp -r resources/ dist"
             "npx shadow-cljs watch :plugin :test"]]
  (shell cmd))
```
Maybe a `shells` helper? LOL
A matter of personal preference anyway

**Reply by rlhk** [reply]
Wrote a simple fn to make `bb.edn` a bit more concise:
https://github.com/rlhk/logseq-url-plus/blob/main/bb.edn
